streaming_jupyter_integrations/magics.py

```diff
@@ -76,7 +67,104 @@ def __init__(self, shell: Any):
+    @argument("-yid", "--yarn-application-id", type=str, help="Flink Session Cluster applicationId", required=False)
+    def flink_connect(self, line: str) -> None:
+        args = parse_argstring(self.flink_connect, line)
+        try:
+            self._flink_connect(args)
+        except ValueError as e:
+            print(e, file=sys.stderr)
+
+    def _flink_connect(self, args: Any) -> None:
+        execution_mode = args.execution_mode
+        execution_target = args.execution_target
+
+        if execution_target == "local":
+            self._flink_connect_local()
+        elif execution_target == "remote":
+            self._flink_connect_remote(args.remote_hostname, args.remote_port)
+        elif execution_target == "yarn-session":
+            self._flink_connect_yarn_session(args.resource_manager_hostname, args.resource_manager_port,
+                                             args.yarn_application_id)
+        else:
+            raise ValueError(
+                f"Unknown execution mode. Expected 'local', 'remote' or 'yarn-session', actual '{execution_target}'.")
+
         self.__flink_execute_sql_file("init.sql")
+        self._set_table_env(execution_mode)
+        print(f"{execution_target} environment has been created.")
+
+    def _flink_connect_local(self) -> None:
+        global_conf = self.__read_global_config()
+        conf = self.__create_configuration_from_dict(global_conf)
+        conf.set_integer("rest.port", 8099)
+        conf.set_integer("parallelism.default", 1)
+        self.s_env = StreamExecutionEnvironment(
+            get_gateway().jvm.org.apache.flink.streaming.api.environment.StreamExecutionEnvironment.createLocalEnvironmentWithWebUI(  # noqa: E501
+                conf._j_configuration
+            )
+        )
+
+    def _flink_connect_remote(self, hostname: str, port: int) -> None:
+        gateway = get_gateway()
+        empty_string_array = gateway.new_array(gateway.jvm.String, 0)
+        self.s_env = StreamExecutionEnvironment(
+            gateway.jvm.org.apache.flink.streaming.api.environment.StreamExecutionEnvironment.createRemoteEnvironment(
+                hostname, port, empty_string_array
+            )
+        )
+
+    def _flink_connect_yarn_session(self, rm_hostname: str, rm_port: int, yarn_application_id: str) -> None:
+        jm_hostname, jm_port = self._find_session_jm_address(rm_hostname, rm_port, yarn_application_id)
+        self._flink_connect_remote(jm_hostname, jm_port)
+
+    def _find_session_jm_address(self, rm_hostname: str, rm_port: int, yarn_application_id: str) -> Tuple[str, int]:
+        if rm_hostname and rm_port:
+            rm_client = ResourceManagerClient(rm_hostname, rm_port)
+        else:
+            rm_client = ResourceManagerClient()
+        if yarn_application_id is None:
+            yarn_application_id = self._find_yarn_application_id(rm_client)
```

Comment:
I would wrap the parameters in the `Optional[...]` type annotation.

Reply:
done